src/app/app.component.ts

```diff
@@ -11,7 +12,13 @@ import {UserService} from './services/user.service';
 export class AppComponent {
   title = 'Angular SpringBoot JWT integration';
 
-  constructor(private router: Router, private userService: UserService) {
+  constructor(private router: Router, private userService: UserService, private cdRef:ChangeDetectorRef) {
+
+  }
+
+  ngAfterViewChecked() { 
```

Comment:
can you please explain why this is required? what is failing?

Reply:
That code just avoid the app to throw an error (seen in the console) related to `ExpressionChangedAfterItHasBeenCheckedError` when navigating to the user component. You can find more info in [here](https://github.com/angular/angular/issues/17572#issuecomment-309229619)